### PR TITLE
Add BOM to CsvWriter

### PR DESCRIPTION
**Pull request by ThibautAlgrin**

### CODE REVIEW

File: lib/Exporter/Writer/CsvWriter.php [hunk 1]
```diff
@@ -29,21 +29,25 @@ class CsvWriter implements WriterInterface
...
 
+    protected $hasBom = null;
+
     /**
      * @param string $filename
      * @param string $delimiter
      * @param string $enclosure
      * @param string $escape
      * @param bool   $showHeaders
+     * @param bool   $hasBom
```
Comment (rande) [inline lib/Exporter/Writer/CsvWriter.php]: maybe ``withBom`` is more adequate

**Comment by rande** [comment]
Can you add a new unit test ?

File: lib/Exporter/Writer/CsvWriter.php [hunk 2]
```diff
@@ -29,21 +29,25 @@ class CsvWriter implements WriterInterface
 
     protected $position;
 
+    protected $hasBom = null;
```
Comment (Soullivaneuh) [inline lib/Exporter/Writer/CsvWriter.php]: Should not have default value as it's set on constructor.

BTW, default values are different (`null`/`false`).

File: lib/Exporter/Writer/CsvWriter.php [hunk 3]
```diff
@@ -29,21 +29,25 @@ class CsvWriter implements WriterInterface
...
+     * @param bool   $hasBom
      */
-    public function __construct($filename, $delimiter = ',', $enclosure = '"', $escape = '\\', $showHeaders = true)
+    public function __construct($filename, $delimiter = ',', $enclosure = '"', $escape = '\\', $showHeaders = true, $hasBom = false)
     {
-        $this->filename    = $filename;
-        $this->delimiter   = $delimiter;
-        $this->enclosure   = $enclosure;
-        $this->escape      = $escape;
+        $this->filename = $filename;
```
Comment (Soullivaneuh) [inline lib/Exporter/Writer/CsvWriter.php]: Please do not change not concerned code.

File: lib/Exporter/Writer/CsvWriter.php [hunk 4]
```diff
@@ -56,6 +60,9 @@ public function __construct($filename, $delimiter = ',', $enclosure = '"', $esca
     public function open()
     {
         $this->file = fopen($this->filename, 'w', false);
+        if ($this->hasBom) {
```
Comment (Soullivaneuh) [inline lib/Exporter/Writer/CsvWriter.php]: `true === $this->hasBom`

File: lib/Exporter/Writer/CsvWriter.php (hunk 2 above)
Comment (Soullivaneuh) [inline lib/Exporter/Writer/CsvWriter.php]: Missing php doc block.

**Comment by ThibautAlgrin** [comment]
I made the wanted modifications. Can you validate Pull Request?

**Comment by Soullivaneuh** [comment]
Commit need to be squashed.

I let @rande review and validate this PR after that.

**Comment by ThibautAlgrin** [comment]
What do you want me to do?

**Comment by Soullivaneuh** [comment]
> What do you want me to do?

To squash your commits onto one.

**Comment by Soullivaneuh** [comment]
@ThibautAlgrin Next time, use `fixup` instead of `squash` to avoid extra comment on your commit https://github.com/ThibautAlgrin/exporter/commit/df5a2dc5f2e539887a0f24e558c017953f262e62 ;-)

**Comment by OskarStark** [comment]
@Soullivaneuh is this RTM?

**Comment by OskarStark** [comment]
or @rande can decide this, too

**Comment by rande** [comment]
yes it is, thanks

**Comment by OskarStark** [comment]
Thank you @ThibautAlgrin

**Pull request merged**